ui/src/main/kotlin/org/jetbrains/jewel/ui/component/ListComboBox.kt

```diff
@@ -76,6 +78,82 @@ public fun ListComboBox(
         selectedItem = selectedItem.minus(1).coerceAtLeast(0)
         scope.launch { scrollState.lazyListState.scrollToIndex(selectedItem) }
     }
+    val onEnterPress = {
+        val indexOfSelected = items.indexOf(inputTextFieldState.text)
+        if (indexOfSelected != -1) {
+            selectedItem = indexOfSelected
+        }
+    }
+
+    fun onSelectedIndexesChange() = { selectedItems: List<Int> ->
+        if (selectedItems.isNotEmpty()) {
+            val selectedItemIndex = selectedItems.first()
+            selectedItem = selectedItemIndex
+            inputTextFieldState.setTextAndPlaceCursorAtEnd(items[selectedItemIndex])
+            onSelectedItemChange(items[selectedItemIndex])
+        }
+    }
+
+    fun contentItems(
+        items: List<String>,
+        onHoverItemChange: (String) -> Unit,
+        listItemContent: @Composable (String, Boolean, Boolean, Boolean, Boolean) -> Unit,
+    ): SelectableLazyListScope.() -> Unit {
+        return {
+            items(
+                items = items,
+                itemContent = { item ->
+                    var isItemHovered by remember { mutableStateOf(false) }
+                    Box(
+                        modifier =
+                            Modifier.onHover {
+                                isItemHovered = it
+                                if (isItemHovered) {
+                                    hoverItemIndex = items.indexOf(item)
+                                    onHoverItemChange(item)
+                                }
+                            }
+                    ) {
+                        listItemContent(
+                            item,
+                            isSelected,
+                            isActive,
+                            isItemHovered || items.indexOf(item) == hoverItemIndex,
+                            hoverItemIndex != null,
+                        )
+                    }
+                },
+            )
+        }
+    }
+
+    @Composable
+    fun list(
+        items: List<String>,
+        scrollState: SelectableLazyListState,
+        popupMaxHeight: Dp,
+        onListHoverChange: (Boolean) -> Unit,
+        onHoverItemChange: (String) -> Unit,
+        listItemContent: @Composable (String, Boolean, Boolean, Boolean, Boolean) -> Unit,
+    ) {
+        VerticallyScrollableContainer(
+            scrollState = scrollState.lazyListState,
+            modifier =
+                Modifier.heightIn(max = popupMaxHeight).onHover {
+                    isListHovered = it
+                    onListHoverChange(it)
+                },
+        ) {
+            SelectableLazyColumn(
+                modifier =
+                    Modifier.fillMaxWidth().heightIn(max = popupMaxHeight).padding(horizontal = 6.dp, vertical = 6.dp),
```

Comment:
I'm aware this was not introduced here, but are you sure this padding should not be a contentPadding instead? And, should it be in the styling?

```suggestion
                    Modifier.fillMaxWidth().heightIn(max = popupMaxHeight).padding(6.dp),
```

Reply:
I'm in favor of extracting it into the style: https://github.com/JetBrains/jewel/pull/669/commits/edf0314170214d6416e1dcbe891d355354743f17

The problem that I have now comes from the ListComboBox being a bit “arbitrary” when it comes to metrics because all the comboboxes in UI Sandbox, Preferences, Figma are different: I can't find that `6.dp` we are using in the IntelliJ source code 😬
Any suggestion is welcome 🙏🏻

Reply:
As usual, my approach is at a high level to try and get to this 6.dp from values we can obtain from the LaF; however, if they don't exist in the LaF, and these are hardcoded in Swing, even as sum of other hardcoded values, it's ok to hardcode them on our side too.

If they come in any way shape or form from the LaF in Swing, then they need to be constructed from the same values (e.g., a LaF value plus a fixed value)